### DISCUSSION TITLE
Gurobi Environment handling

### DISCUSSION THREAD

**Discussion by torressa** [google/or-tools#3633] [Feature requests]
**What language and solver does this apply to?**
Every language except C++ (this already exists).

**Describe the problem you are trying to solve.**

I would like to be able to pass a Gurobi environment as input. 
In C++ this already exists via https://github.com/google/or-tools/blob/82750ac12f1ee5354e1c7869894d9af3508778f2/ortools/linear_solver/gurobi_proto_solver.h#L35-L36

All other APIs cannot take a user-defined environment (Gurobi has an `Env` construct for all Python, Java, .NET).
Please correct me if this is already possible.

**Describe the solution you'd like**

To be able to pass an environment for example

```python
import gurobipy as gp

with gp.Env() as myenv:
  solver = pywraplp.Solver.CreateSolver("GUROBI", env=myenv)
```

**Comment by lperron** [comment]
This is not possible.

**Reply by ronaldvdv** [reply]
Currently it's not possible - but what would need to change to make this possible? @lperron Do you have specific concerns around implementing this possibility for languages like Python?

**Comment by lperron** [comment]
Very complex. I do not want to depend on gurobipy, and the code is not
structured this way.

The only solution is to pass the structure that holds the isv key from
python to c++.
